cmd/finish/main_test.go

```diff
@@ -0,0 +1,193 @@
+package main
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/opendevstack/pipeline/pkg/config"
+	"github.com/opendevstack/pipeline/pkg/logging"
+	"github.com/opendevstack/pipeline/pkg/nexus"
+	"github.com/opendevstack/pipeline/pkg/pipelinectxt"
+)
+
+func TestUploadArtifacts(t *testing.T) {
+	logger := &logging.LeveledLogger{Level: logging.LevelDebug}
+	nexusRepo := "foo"
+	nexusClient := &nexus.TestClient{
+		URLs: map[string][]string{},
+	}
+	tempWorkingDir, cleanup, err := prepareTempWorkingDir(nexusRepo)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer cleanup()
```

Comment:
I think as `prepareTempWorkingDir(nexusRepo)` could fail both, before having created the tempdir and afterwards (when calling into `writeArtifactsManifest(...)`), we'd have to call `cleanup()` no matter what before failing the test case, otherwise the tempdir might stick around. Placing the defer before the error check should fix this IMO.

Reply:
Great catch, thanks for pointing this out. Fixed in db117fb.